### PR TITLE
[Model] add warning if flat-map is used

### PR DESCRIPTION
**Pull request by philbucher**
This will give a warning that one should always give the full name to retrieve a ModelPart from the Model.
I noticed that many devs still don't know abt this, this is a first step of telling them.

For now it is only a `KRATOS_INFO`, not yet a `KRATOS_WARNING`, this way it will not yet show up in the tests
If you prefer we can also make it a `KRATOS_WARNING` right away

To be merged after the release, such that we won't have warnings everywhere

### CODE REVIEW

**Comment by philbucher** [comment]
ping, release branch is made :)

**Comment by philbucher** [comment]
ping

**Comment by pooyan-dadvand** [comment]
Sorry @philbucher, we have missed this one yesterday. We will discuss it in the next meeting

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee agrees with adding the warning but not convinced about the name of `RecursiveGetFullName`. We would remove the Recursive and something more descriptive rather than GetFullName (like GetNameWithAscendants or similar options to be found by some native English speaker) would be better.

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee I can ofc update the function name, but note that this function is temporary and will be removed once the flatmap is removed

**Comment by philbucher** [comment]
done, name is changed to `GetNameWithAscendants`

**Comment by pooyan-dadvand** [comment]
Why this function should be remove in future?

**Comment by philbucher** [comment]
I will no longer be needed
It is only used if the input is given in the deprecated format, i.e. a submodelpart-name without it's parent
=> e.g. if the user gives `Inlet` the correct/non-deprecated name would be `FluidModelPart.Inlet` then I use this function to get the name of the parent-modelpart
I use the function ONLY to say in the warning how the correct input should be, see [here](https://github.com/KratosMultiphysics/Kratos/blob/model/deprecate-flat-map/kratos/containers/model.cpp#L116-L119)

File: kratos/containers/model.cpp [hunk 1]
```diff
@@ -98,11 +98,28 @@ namespace Kratos
...
+                        std::stringstream msg;
+                        msg << model_part_names[0];
+                        for (std::size_t i=1; i<model_part_names.size(); ++i) {
+                            msg << "." << model_part_names[1];
+                        }
+
+                        KRATOS_INFO("Model") << "DEPREATION_WARNING: The ModelPart \"" // TODO make a warning at some point
+                            << subparts_list[0] << "\"\nis retrieved from the Model by using the "
+                            << "flat-map!\nPlease prepend the Parent-ModelPart-Names like this:\n\""
+                            << msg.str() << "\"" << std::endl;
```
Comment (philbucher) [inline kratos/containers/model.cpp]: here I tell the user how the correct name should be

**Comment by pooyan-dadvand** [comment]
Thanks, now I see.

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee agrees. Thanks!

**Comment by philbucher** [comment]
can someone approve then? Thx

**Review by loumalouomega — APPROVED** [review]

**Comment by philbucher** [comment]
thx

**Pull request merged**